timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -532,19 +538,18 @@ void __global__ k_compute_nonbonded_target_atom_energies(
                             1.0, eps_i, eps_j, sig_i, sig_j, inv_dij, inv_d2ij, u, delta_prefactor, sig_grad, eps_grad);
                     }
                     // Store the atom by atom energy
-                    block_energy_buffer[threadIdx.x] = FLOAT_TO_FIXED_ENERGY<RealType>(u);
+                    energy_accumulator = FLOAT_TO_FIXED_ENERGY<RealType>(u);
                 }
             }
-            // Sync to ensure the shared buffers are populated
-            __syncthreads();
 
-            block_energy_reduce<THREADS_PER_BLOCK>(block_energy_buffer, threadIdx.x);
+            // Sum's return value is only valid in thread 0
```

Comment:
interesting switch of the location of the `__syncthreads()` - from before the reduce to after the reduce - intentional?

Reply:
i.e. guessing that BlockReduce::Sum() implicitly calls a syncthreads